plugin/src/main/java/org/gradle/testretry/internal/config/TestRetryTaskExtensionAccessor.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.testretry.internal.config;
+
+import java.util.Set;
+
+public interface TestRetryTaskExtensionAccessor {
```

Comment:
I'm not sure whether this should outside the internal package, as we will use this interface.

At the moment, we only have:
* `TestRetryPlugin`
* `TestRetryTaskExtension`

on the top level. All the rest is inside an `internal` package.

Should the read-only interface for the extension also be outside?WDYT?

Reply:
I don't mind having it inside the `internal` package. I guess we do not expect customers to interact with it, do we?

Reply:
No, we don't expect customers to use it. The decoupling is really just meant for us, so we can use the `TestRetryExecuter` in Gradle Enterprise with a different implementation of the `TestRetryTaskExtension`.